src/pages/ScreenRouting/mediaScreens/MediaSeries/MediaSeries.tsx

```diff
@@ -1,21 +1,48 @@
+import { Navigate } from 'react-router';
+import { useTranslation } from 'react-i18next';
+
 import type { ScreenComponent } from '#types/screens';
-import { getSeriesId } from '#src/utils/media';
-import SeriesRedirect from '#src/containers/SeriesRedirect/SeriesRedirect';
 import type { PlaylistItem } from '#types/playlist';
+import { useSeriesData } from '#src/hooks/useSeriesData';
+import useQueryParam from '#src/hooks/useQueryParam';
+import { episodeURL } from '#src/utils/formatting';
 import Loading from '#src/pages/Loading/Loading';
+import ErrorPage from '#components/ErrorPage/ErrorPage';
+import { getSeriesIdFromCustomParams } from '#src/utils/media';
 
 /**
- * This media screen is used to redirect a series linking media item to an episode page.
+ * This container is mainly used to redirect the user to the correct episode media page.
+ *
+ * It fetches the series and redirects to the first episode.
+ *
+ * Later, the watch history can also be used to determine the last watched episode to properly support continue
+ * watching with series.
  */
-const MediaSeries: ScreenComponent<PlaylistItem> = ({ data, isLoading }) => {
-  const seriesId = getSeriesId(data) || '';
+const MediaSeries: ScreenComponent<PlaylistItem> = ({ data: media, isLoading: isFetchingMedia }) => {
+  const seriesId = getSeriesIdFromCustomParams(media);
+  const mediaId = media.mediaid;
+
+  const { t } = useTranslation('video');
 
-  // prevent rendering the SeriesRedirect multiple times when we are loading data
-  if (isLoading) {
+  const { isLoading, isPlaylistError, data } = useSeriesData(seriesId, mediaId);
+  const play = useQueryParam('play') === '1';
+  const feedId = useQueryParam('r');
+
+  if (isLoading || isFetchingMedia) {
     return <Loading />;
   }
 
-  return <SeriesRedirect seriesId={seriesId} />;
+  if (isPlaylistError || !data) {
+    return <ErrorPage title={t('series_error')} />;
+  }
+
+  const toEpisode = data.seriesPlaylist.playlist[0];
+
+  if (!toEpisode) {
+    return <ErrorPage title={t('episode_not_found')} />;
+  }
+
+  return <Navigate to={episodeURL({ episode: toEpisode, seriesId: seriesId || mediaId, mediaId, play, playlistId: feedId })} replace />;
```

Comment:
What's the reason for this? `seriesId: seriesId || mediaId`

Reply:
Now I pass `seriesId` and `mediaId`, so we can choose which one we need.